tensordict/tensordict.py

```diff
@@ -944,7 +945,11 @@ def expand(self, *shape) -> TensorDictBase:
                         new_shape=shape, old_shape=self.batch_size
                     )
                 )
-        for key, value in self.items():
+
+        keys_view = _TensorDictKeysView(self, include_nested=True, leaves_only=False,
+                                        error_on_loop=False)
+        for key in keys_view:
+            value = self.get(key)
```

Comment:
This doesn't preserve the auto-nesting in the output.

I think this one would be better done with `_apply_safe(lambda _, v: v.expand(...))` etc.

Reply:
Done similarly to below